components/channel_selector_modal/channel_selector_modal.tsx

```diff
@@ -175,7 +185,7 @@ export default class ChannelSelectorModal extends React.PureComponent {
         );
     }
 
-    renderValue(props) {
+    renderValue(props: {data: ChannelWithTeamData}) {
```

Comment:
Should `data` be `ChannelWithTeamData & Value` here?

Reply:
I've updated the MR with this suggestion.